online_meetings/report.py

```diff
@@ -0,0 +1,161 @@
+import requests
+# standard libraries
+import yaml
+import json
+import logging
+import os
+
+from typing import Dict, Union
+
+# third-party libraries
+import pandas as pd
+import time
+import dateparser
+from datetime import datetime, timedelta
+
+# Initialize settings and globals
+
+logger = logging.getLogger(__name__)
+
+# read configurations
+try:
+    with open(os.path.join('../config', 'env.json')) as env_file:
+        ENV = yaml.load(env_file.read())
+except FileNotFoundError:
+    logger.error(
+        'Configuration file could not be found; please add env.json to the config directory.')
+
+logging.basicConfig(level=ENV.get('LOG_LEVEL', 'DEBUG'))
+
+logger.info(ENV)
+
+payload = {}
+
+headers = {
+  "Authorization": f"Bearer {ENV['ZOOM_TOKEN']}"
+}
+
+ZOOM_BASE_URL = ENV.get('ZOOM_BASE_URL', "")
+DEFAULT_SLEEP_TIME = ENV.get('DEFAULT_SLEEP_TIME', "10")
+
+
+# Functions
+def get_total_page_count(url: str, headers: Dict[str, Union[str, int]] = {}, params: dict = {}):
+    # get the total page count
+    total_page_count = 0
+    response = requests.request("GET", f"{url}", headers=headers, params=params)
+    status_code = response.status_code
+    if status_code != 200:
+        logger.warning(f'Received irregular status code: {status_code}')
+        logger.info('No page at all!')
+    else:
+        try:
+            results = json.loads(response.text.encode('utf8'))
+            total_page_count = results['page_count']
+        except json.JSONDecodeError:
+            logger.warning('JSONDecodeError encountered')
+            logger.info('No page at all!')
+    return total_page_count
+
+
+def run_report(api_url: str, headers: dict, json_attribute_name: str,
+               default_params: dict = {}, page_size: int = 300, page_token: bool = False, use_date: bool = False):
+    url = ZOOM_BASE_URL+api_url
```

Comment:
```suggestion
    url = ZOOM_BASE_URL + api_url
```

Reply:
I think I had E226 rule disabled, can enable and fix this.